### PR TITLE
Don't strip all non-debug sections from separate-dwarf file

### PR DESCRIPTION
**Pull request by dschuff**
When LLVM tools such as llvm-dwp and llvm-dwarfdump parse debug info,
they use information from the wasm known sections (such as function
indices) to parse and verify the debug info. This means that #16110
broke use of these tools on separate-dwarf files.

This is basically a revert of #16110 except that it adds tests of
dwp and dwarfdump running on separate-dwarf and split-dwarf files.
We should actually solve #13084 again by stripping out just the code
section (instead of all known sections) but llvm-objdump currently doesn't
know how to do that (it only handles named custom sections currently).
So this PR reopens #13084

### CODE REVIEW

File: tests/test_other.py [hunk 1]
```diff
@@ -8311,16 +8311,28 @@ def test_separate_dwarf(self):
...
+    dwdump = self.run_process(
+      [LLVM_DWARFDUMP, 'subdir/output.wasm.debug.wasm', '-name', 'main'],
+      stdout=PIPE).stdout
+    # Basic check that the debug info is more than a skeleton. If so it will
+    # have a subprogram descriptor for main
+    self.assertIn('DW_TAG_subprogram', dwdump)
+    self.assertIn('DW_AT_name\t("main")', dwdump)
+
+    self.run_process([LLVM_DWP, '-e', 'subdir/output.wasm.debug.wasm',
+                      '-o', 'subdir/output.wasm.debug.wasm.dwp'])
```
Comment (kripken) [inline tests/test_other.py:8334]: are there any checks we should do on the output file here? Or is this command just run to check for not crashing? (if so do we need the `-o`?)

Comment (dschuff) [reply tests/test_other.py:8334]: Ah, it was originally to check for not crashing (which is the current symptom) but probably it makes sense to run dwarfdump over the output too.

... And I just realized dwp really only makes sense with -gsplit-dwarf, which we don't have an end-to-end emscripten test for. So I added that.

File: tests/test_other.py [hunk 2]
```diff
@@ -8311,15 +8311,51 @@ def test_separate_dwarf(self):
...
       self.fail('name section not found in separate dwarf file')
     for sec in debug_wasm.sections():
-      if sec.type != webassembly.SecType.CUSTOM:
-        self.fail(f'non-custom section type {sec.type} found in separate dwarf file')
-      elif sec.name != 'name' and not sec.name.startswith('.debug'):
+      # TODO: check for absence of code section (see
+      # https://github.com/emscripten-core/emscripten/issues/13084)
+      if sec.name and sec.name != 'name' and not sec.name.startswith('.debug'):
         self.fail(f'non-debug section "{sec.name}" found in separate dwarf file')
+    shared.PRINT_STAGES = True
```
Comment (kripken) [inline tests/test_other.py:8323]: leftover debugging?

**Review by kripken — APPROVED** [review]

**Comment by dschuff** [comment]
OK, assuming the tests pass, I'll land this.
@pfaffe I noticed that when this tests runs `llvm-dwarfdump` over a .dwp file, I'm getting some warnings:
```
warning: Unexpected relocations for dwo section debug_abbrev.dwo
warning: Unexpected relocations for dwo section debug_str.dwo
warning: Unexpected relocations for dwo section debug_str_offsets.dwo
warning: Unexpected relocations for dwo section debug_info.dwo
```
Do you see something like this on whatever version of emscripten you're currently using/testing with? If so, I assume it still works in the debugger?

**Comment by dschuff** [comment]
Firefox test failure seems unrelated, I'm going to go ahead and land this, since I think it's working now, and it was definitely broken before. Still interested in hearing from @pfaffe though.

**Pull request merged**